File: 01_Language/01_Functions/python/pathinfo.py
# coding: utf-8

from os import path


def pathinfo(p):
    filename, ext = path.splitext(path.basename(p))
    return {
        'dirname': path.dirname(p),
        'basename': path.basename(p),
        'extension': ext,
        'filename': filename,
    }


if __name__ == '__main__':
    print(pathinfo('./text.txt'))
    print(pathinfo('./text'))
    print(pathinfo('./.a'))
    # print(path.splitext('./.a'))

File: 01_Language/01_Functions/php/pathinfo.php
<?php

$path_parts = pathinfo('./test.txt');

var_dump($path_parts);
//echo $path_parts['dirname'], "\n";
//echo $path_parts['basename'], "\n";
//echo $path_parts['extension'], "\n";
//echo $path_parts['filename'], "\n"; // since PHP 5.2.0
